### PR TITLE
Add node stats effective watermark thresholds docs.

### PR DESCRIPTION
**Pull request by howardhuanghua**
In https://github.com/elastic/elasticsearch/pull/107244 we added effective watermark stats in node stat API, this PR update related docs.

### CODE REVIEW

File: docs/reference/cluster/nodes-stats.asciidoc [hunk 1]
```diff
@@ -1806,6 +1806,38 @@ Total number of bytes available to this Java virtual machine on all file
 stores. Depending on OS or process level restrictions (e.g. XFS quotas), this might appear
 less than `free_in_bytes`. This is the actual amount of free disk
 space the {es} node can utilise.
+
+`low_watermark_free_space`::
+(<<byte-units,byte value>>)
+Total amount of effective low watermark free disk space available to this Java virtual machine on this file store.
```
Comment (DaveCTurner) [inline docs/reference/cluster/nodes-stats.asciidoc:1812]: Wording suggestion

```suggestion
The effective low disk watermark for this data path on this node: when a node
has less free space than this value for at least one data path, its disk usage
has exceeded the low watermark. See <<disk-based-shard-allocation>> for more
information about disk watermarks and their effects on shard allocation.
```

**Review by DaveCTurner — COMMENTED** [review]
I left a suggested wording with a little more detail. Also these docs should go within the `data` subsection since they are per-data-path things. And the other references to "file store" should probably instead refer to "data path" (or maybe "file store containing the data path") since we don't really talk about file stores elsewhere.

**Review by DaveCTurner — COMMENTED** [review]
Oh yes and no need for a changelog entry for docs changes, this check will have gone away when I added the `>docs` label

File: docs/reference/cluster/nodes-stats.asciidoc (hunk 1 above)
Comment (howardhuanghua) [reply docs/reference/cluster/nodes-stats.asciidoc:1812]: OK, I will update other threshold types based on the suggestion, and also add them to `data` subsection.

File: docs/reference/cluster/nodes-stats.asciidoc [hunk 2]
```diff
@@ -1853,6 +1853,51 @@ store.
...
+`low_watermark_free_space`::
+(<<byte-units,byte value>>)
+The effective low disk watermark for this data path on this node: when a node
+has less free space than this value for at least one data path, its disk usage
+has exceeded the low watermark. See <<disk-based-shard-allocation>> for more
+information about disk watermarks and their effects on shard allocation.
+
+`low_watermark_free_space_in_bytes`::
+(integer)
+The effective low disk watermark in bytes for this data path on this node.
```
Comment (DaveCTurner) [inline docs/reference/cluster/nodes-stats.asciidoc:1866]: May as well include the same info for the `in_bytes` variants:

```suggestion
The effective low disk watermark, in bytes, for this data path on this node:
when a node has less free space than this value for at least one data path, its
disk usage has exceeded the low watermark. See <<disk-based-shard-allocation>>
for more information about disk watermarks and their effects on shard
allocation.
```

**Review by DaveCTurner — APPROVED** [review]
LGTM

**Comment by DaveCTurner** [comment]
@elasticmachine test this please

**Pull request merged**